### DISCUSSION TITLE
Is --only supposed to respect overrides?

### DISCUSSION THREAD

**Discussion by chmmpagne** [biomejs/biome#4156] [Q&A] [answered]
I was using `--only` to look at a subset of issues as well as only running a fixer for a single rule.

I got a bit confused when I noticed cases where I had turned the rule off for certain files and was still getting diagnostics when using `--only`.
Omitting `--only` and piping the diagnostics to a file (`biome lint --max-diagnostics=none &> biome.log`) allowed me to see that these rules were being turned off correctly.

Just wondering if ~~this is intentional,~~ there's another way to do this ~~, or this is a bug~~.

**Comment by chmmpagne** [comment]
Oh, I just noticed in the docs ...

> --only=<GROUP|RULE> — Run only the given rule or group of rules. If the severity level of a rule is off, then the severity level of the rule is set to error if it is a recommended rule or warn otherwise.

I'd still be curious if there's a way to actually run only the given rules while respecting the configuration (i.e., not automatically changing severity levels).

**Comment by ematipico** [comment, marked as answer]
The CLI options usually have higher precedence over `biome.json`, so no, it isn't possible to achieve what you desire.

However, you can use `--skip` as a workaround

**Comment by Conaclos** [comment]
I understand your problem, but I don't see how we can support your use case in a consistent and intuitive way.

**Comment by chmmpagne** [comment]
Thanks for quick replies!
Unfortunate, but yeah I'm not sure how you'd support it properly off the top of my head.
I'll look into using `--skip` or another solution.
